### DISCUSSION TITLE
Graphing with Task based model

### DISCUSSION THREAD

**Discussion by wilsoncg** [microsoft/coyote#404] [General]
Hello,
I watched the On .NET show coyote videos, great work on those. You should definitely add a link to those on documentation pages.

I had a look at the bounded buffer sample and played around with some code of my own using the Task model. I tried out the --graph option, but noticed nothing is output (only some skeleton styling). I noticed this mentioned on the gitter, that the graph option is more geared to the actor model. It would be really great if you could provide the additional logging code that was used in the video to generate the chart, would you be able to add that part to the sample code? @lovettchris 

I notice the readme was removed from the bounded buffer example, why was it removed? I'm guessing it was supposed to be moved to the documentation site but that step was forgotten?

**Comment by pdeligia** [comment]
Hi @wilsoncg, thanks for your interest in Coyote!

Yes, that's a great suggestion, we indeed plan to add a link to those videos on our website ASAP.

Regarding the `--graph` option for producing DGML diagrams, you are right that its currently only supported for actor-based programs, as it's using the underlying actor logger (which gives us all actor events e.g. state transitions, message sends, actor creations, etc, which we then use for visualization), this is something that we are investigating how to visualize for task programs. @lovettchris did the actor DGML feature in Coyote, and he is away this week, so please expect some delay in responses.

We removed the readme from `BoundedBuffer` because we published it as a blog post here https://cloudblogs.microsoft.com/opensource/2020/07/14/extreme-programming-meets-systematic-testing-using-coyote/. But we are thinking to create a similar "walkthrough" documentation page to our other samples, and add it in the website, which will then also link to our blog post, or just directly link to the blog post.
